### ISSUE TITLE
Not working

### ISSUE THREAD

**Issue by Nephrite-FM** [jtgans/PS2FF#1]
I have tried to assemble this, however it is not working correctly.  When inserted the drive isn't recognized at all.

I'll try putting together another board, and I notice that you do not indicate on the board the orientation of the 74LS125.

Is there any news on Revision 3?

**Comment by jtgans** [comment]
Ah, apologies. Yeah, it's not working right. My next steps are to attach my network analyzer to the bus and figure out the timing diagrams.

On my PS/2 Model 77, the controller would recognize the drive, but only as a 5.25" floppy drive, instead of a 3.5". I'm not sure exactly why this occurs, so I suspect there's more going on in the protocol than what is documented in IBM's enhanced floppy drive documentation. Simple static line settings aren't enough, apparently. Definitely need timing diagrams of what happens on the secure, bitrate, and drive type lines during system initialization to better understand what is going on.

**Comment by jtgans** [comment]
Didn't mean to close this.

Revision 3 is under development, been on hold until I can get back to it. Ran out of time with VCFSW, so that had to take priority. Will probably return to this very soon.

**Comment by jtgans** [comment]
Orientation of the 74LS125 is using KiCad's braindead silk screen patterning, so use that as your guide.

**Comment by SudosFTW** [comment]
> Revision 3 is under development, been on hold until I can get back to it. Ran out of time with VCFSW, so that had to take priority. Will probably return to this very soon.

I have a 40SX I'd love to pop a Gotek into and stumbled upon your project... and as such I await Rev 3 whenever that occurs. I have a working floppy drive already (hint: Y-E Data PS/2 floppy drives don't go caps-up!) but as floppies are getting scarce... yeah.

However!
It looks like a user handled Panagis1980 on eBay is selling their own Media Sense adapter already, and the design is [also up on PCBWay.](https://www.pcbway.com/project/shareproject/IBM_PS_2_Floppy_adapter_v_1_for_Media_Sense_cables_34c4067f.html) For all that I can see, it's a 100% rehash of your project but laid out differently? right down to the 74LS125 tri-state buffer... but people have supposedly gotten their adapter to work if the feedback on eBay is to be believed, and there's even a video showing it working right in [the auction page.](https://www.ebay.com/itm/186188234654)

_EDIT: it looks like the one on PCBWay is Rev B, they're selling Rev A boards on eBay. the difference seems to be a jumper for density inversion._

While theirs is ready-to-be-bought assembled and one could just get the BOM from pcbway and have them assemble it, I'd rather go with your project as you have proper kicad files and a proper digikey cart already set to go. The other person has only shared it to pcbway, and while that has a BOM and everything, it doesn't look like the project files themselves are downloadable or modifiable? Kinda scummy, especially considering the website on the PCB itself is a not-set-up Microsoft Azure instance. Just wanting to make sure you know about it is all. 

![s-l1600](https://github.com/jtgans/PS2FF/assets/11806373/938f36e7-c0bb-438d-b61b-4291043bd20a)

For the record, is the reason yours doesn't work only because it's stuck in 1.44M mode like that possible clone board, or what's the deal? if anything I'd go and order one of your Rev 2 boards now from osh if it behaves as expected and then just transfer components over whenever Rev 3 ends up happening.

**Comment by jtgans** [comment]
Yeah, I'm not sure what that guy is selling, TBH. The photos of the board layout on PCBWay show a Q1 -- possibly a transistor or an LDO in there or something, and the pads don't look correct -- some of those resistor footprints look positively *weird*. None of my board revisions have that Q1 anywhere in the design. Weird that he's essentially copied my revision scheme and many of my texts in the silks -- guess he used mine as something to base his off of.

I think we can call this parallel invention. I don't see any particular problems here. His use case is different as well. I wanted the PS2FF to be an extension of the Gotek to allow it to report density properly to the host controller (so we can use images of refdisks rather than being stuck without), whereas his is a fixed density report to the machine.

I should note that in a modern-ish PS/2, having a 1.44MB floppy is *not* enough to load a refdisk, so if you use his adapter, you will likely be unable to change your MCA settings when it detects a misconfig or system error.

No, the reason mine doesn't work is because, well, it just _doesn't work_. In the prototypes I've built up, it makes my Model 77 think it has a 360KB 5.25" floppy drive attached. There's a sequencing problem going on in my 77 that makes it misunderstand what is attached. It's going to take some examination of the timings for multiple signals to figure this out. I'll see if I can maybe get some time on my Saleae today.

**Comment by jtgans** [comment]
> I have a 40SX I'd love to pop a Gotek into and stumbled upon your project... and as such I await Rev 3 whenever that occurs. I have a working floppy drive already (hint: Y-E Data PS/2 floppy drives don't go caps-up!) but as floppies are getting scarce... yeah.

What's the type number, 8540? I don't know of any desktop PS/2s that carried the "40SX" moniker -- do you mean the L40SX laptop?

**Comment by SudosFTW** [comment]
> What's the type number, 8540? I don't know of any desktop PS/2s that carried the "40SX" moniker -- do you mean the L40SX laptop?

Correct, 8540-045. Nameplate says 40SX. 8535/8540 shared a motherboard between two case types and they're both Media Sense machines. The L40SX is a completely different beast altogether.

The model 40 is what one would call rare, I guess? They're not a very common model, but the 35 is a bit more common. 

[Ardent-Tool information page on the model here.
](https://www.ardent-tool.com/qtechinfo/GJAN-43VKWC.html)
[Same site, planar info page for 35/40](https://www.ardent-tool.com/35_40/Planar.html)

**Comment by jtgans** [comment]
Hah -- wild. I was looking over ardent-tool to see if I could find it. Guess my search-fu just isn't there yet. :P

I have an L40SX here, FWIW -- it was my first laptop ever. :D

**Comment by SudosFTW** [comment]
> Yeah, I'm not sure what that guy is selling, TBH. The photos of the board layout on PCBWay show a Q1 -- possibly a transistor or an LDO in there or something, and the pads don't look correct -- some of those resistor footprints look positively _weird_. None of my board revisions have that Q1 anywhere in the design. Weird that he's essentially copied my revision scheme and many of my texts in the silks -- guess he used mine as something to base his off of.
> 
> I think we can call this parallel invention. I don't see any particular problems here. His use case is different as well. I wanted the PS2FF to be an extension of the Gotek to allow it to report density properly to the host controller (so we can use images of refdisks rather than being stuck without), whereas his is a fixed density report to the machine.
> 
> I should note that in a modern-ish PS/2, having a 1.44MB floppy is _not_ enough to load a refdisk, so if you use his adapter, you will likely be unable to change your MCA settings when it detects a misconfig or system error.
> 
> No, the reason mine doesn't work is because, well, it just _doesn't work_. In the prototypes I've built up, it makes my Model 77 think it has a 360KB 5.25" floppy drive attached. There's a sequencing problem going on in my 77 that makes it misunderstand what is attached. It's going to take some examination of the timings for multiple signals to figure this out. I'll see if I can maybe get some time on my Saleae today.

Ah, it looked almost like a direct copy but I'm 100% sure it's not now after following some of the traces from the 74LS125 to where they go. the one on their board is absolutely wired up differently. the PCBWay board is indeed a newer revision with what might be a transistor, not sure myself. I don't have an account with them to go and check the BOM. The jumper for straight-through and inversion is also a questionable addition I'm unsure of...

but it's a race to be first for full functionality I guess. you're one of three I now know working on such an adapter, seems eBay user Markgm is also supposedly working on such a design, or has at least stated it on another listing that one is "coming soon", but I don't know when that was placed there.

In my case, 35/40 are ISA machines like the 25/30 before it and might be able to use the one on eBay/pcbway but eh. 
but anything below model 50 is ISA while 50 and above are all MCA, for the record. ISA machines have a starter disk that does not require ADF files, just plug and go like any normal PC. They also handle the floppies a bit differently at boot for that reason, but the 35/40 were definitely supporting of the 2.88MB drives.

As far as timings, might be an idea to try using the HCT part instead of the LS part instead? Also has the benefit of being a cheaper part, as LS 7400 chips are starting to get rather expensive. I know some things with older machines like the XT-IDE board don't like HCT parts for some of the logic (as told to me in person by glitchwrks at VCF East a couple years back), but I doubt it'll be any issue with floppy. in fact I'm sure the LS part is more than fast enough... but testing HCT part compatibility may help in the diagnostic stage while it's still being questioned.

**Comment by vibr77** [comment]
Based on your very good work, I made this one that is working, I hope it will help others, I think we can change the 74LS125 by a 74LS245 that is a  2 channels OCTAL Bus buffer. https://github.com/vibr77/PS2FloppyDiskAdapterWMediaSense/

**Comment by jtgans** [comment]
Looks like you took that repo private, or at least it doesn't exist yet. What did you change to make it work?

**Comment by jtgans** [comment]
Also, why would we want to use an octal bus buffer instead of a quad? IIRC, we only care about four lines here. TBH, we should be able to do this with just a set of four FETs.

**Comment by vibr77** [comment]
> Looks like you took that repo private, or at least it doesn't exist yet. What did you change to make it work?

My mistake, should be visible

**Comment by vibr77** [comment]
> Also, why would we want to use an octal bus buffer instead of a quad? IIRC, we only care about four lines here. TBH, we should be able to do this with just a set of four FETs.

I meant 74LS244, to avoid using an additional transistor

**Comment by jtgans** [comment]
Wait, where are the actual source files? All you have there are PNGs and gerbers...

**Comment by vibr77** [comment]
It is EasyEDA files for the PCB and the schema
It is at the root of the repo

**Comment by jtgans** [comment]
Also, what's the fix here? You've totally redrawn the schematic. Looks like all you've done is add a transistor to invert the HDOUT pin from the flashfloppy -- is that the only change you made, because I was relying on the flashfloppy firmware to do that part.

**Comment by vibr77** [comment]
Actually, I did only took the idea tu put the 74LS125, I did not check in details your design, and I add a transistor to to invert the HDOUT indeed. This is working like a charm

**Comment by jtgans** [comment]
You're also testing on a model 55, which I think can be faked out with a single switch and a diode. It's nowhere near as complex as the later media sense controllers are, so I suspect this circuit won't work on a 77 or later machine.

Do you normally short JP2 in your use? Because that's the only thing I'm not doing in my schematic.

**Comment by vibr77** [comment]
I have tested it on a 57 model with Media Sense, and JP2 is not shorted, I have add the option in case of need

**Comment by jtgans** [comment]
Ah, yeah, apologies. Model 57, not 55.

I'm going to have to study this more. It's not obvious why yours works and mine doesn't, if JP2 is not shorted when you do your testing. The wiring is identical in that case, aside from the inversion of the HDOUT pin from the flashfloppy. I'm assuming you're leaving JP1 shorted on pins 1 and 2?

**Comment by vibr77** [comment]
yes between 1 and 2, 
do you want my the jumper config of my GOTEK ?

**Comment by jtgans** [comment]
Yes, that would be really helpful, actually! Yeah, maybe I had a software misconfiguration, then, because the only thing I can see different here is that you're just inverting HDOUT.

**Comment by jtgans** [comment]
Bah. I can't use EasyEDA because their interface won't send me an activation email.

**Comment by vibr77** [comment]
JC and S1 are shorted

**Comment by jtgans** [comment]
Okay, so if JC is shorted, what's your FF.CFG say? Are you setting interface = jc or interface = ibmpc?

**Comment by vibr77** [comment]
Interface = jc


From: June Tate-Gans ***@***.***>
Reply to: jtgans/PS2FF ***@***.***>
Date: Friday 8 March 2024 at 15:30
To: jtgans/PS2FF ***@***.***>
Cc: Vincent BESSON ***@***.***>, Comment ***@***.***>
Subject: Re: [jtgans/PS2FF] Not working (Issue #1)


Okay, so if JC is shorted, what's your FF.CFG say? Are you setting interface = jc or interface = ibmpc?

**Comment by jtgans** [comment]
According to https://github.com/keirf/flashfloppy/wiki/FF.CFG-Configuration-File#options if you're using the default FF.CFG, then shorting JC just sets the interface type to `ibmpc`, which leaves pin 2 (HDOUT) low all the time. In the Gotek schematics, it's just driving that signal through a FET which will effectively leave that pin floating if HDOUT is high. In your design, you're effectively always driving that pin before it goes through the transciever. In my design I'm using the pin raw off the FF, which could be floating in the high density case, which would end up indeterminate.

**Comment by jtgans** [comment]
So the reason why yours works is actually because that pin is always being driven low. I'll bet if you set your FF.CFG `interface = ibmpc-hdout`, you'll get different results.

**Comment by vibr77** [comment]
I will do some testing tonight, will come back to you on this one


From: June Tate-Gans ***@***.***>
Reply to: jtgans/PS2FF ***@***.***>
Date: Friday 8 March 2024 at 15:38
To: jtgans/PS2FF ***@***.***>
Cc: Vincent BESSON ***@***.***>, Comment ***@***.***>
Subject: Re: [jtgans/PS2FF] Not working (Issue #1)


So the reason why yours works is actually because that pin is always being driven low. I'll bet if you set your FF.CFG interface = ibmpc-hdout, you'll get different results.

**Comment by jtgans** [comment]
Nope, I'm wrong and missed the pullup on the Gotek. So in my schematic as long as you have `pin02 = ndens` set somewhere in your FF.CFG, the circuit should behave the same way. More digging required.

**Comment by jtgans** [comment]
The more I read, the more I'm thinking your 57 is more tolerant to this schematic than my 77. I'll dig out my board, Saelae, and 77 and 90XP and see what I can see today.

**Comment by vibr77** [comment]
pin02 is auto


From: June Tate-Gans ***@***.***>
Reply to: jtgans/PS2FF ***@***.***>
Date: Friday 8 March 2024 at 15:48
To: jtgans/PS2FF ***@***.***>
Cc: Vincent BESSON ***@***.***>, Comment ***@***.***>
Subject: Re: [jtgans/PS2FF] Not working (Issue #1)


Nope, I'm wrong and missed the pullup on the Gotek. So in my schematic as long as you have pin02 = ndens set somewhere in your FF.CFG, the circuit should behave the same way. More digging required.

**Comment by jtgans** [comment]
Yeah, if it's set to `auto`, then it'll default to the `interface` setting. Since you have `interface` set to `jc`, and you have the JC pin shorted, that means your interface is setup for `ibmpc`, and pin 2 will always be low.

**Comment by vibr77** [comment]
did you try to inverse HDOUT on your design ?

**Comment by jtgans** [comment]
I didn't need to. It's controlled with software.

**Comment by vibr77** [comment]
hum... I would give it a try just to check... trusting soft could sometimes leads to ...

**Comment by jtgans** [comment]
I've read through the software. It shouldn't matter, and definitely shouldn't be causing the issue I was seeing w.r.t. the system mis-identifying the drive as a 1.2MB 5.25" floppy drive.
